week_02/src/clients/updateBudget.js

```diff
@@ -0,0 +1,24 @@
+import Budget from "../models/budget.js";
+import locks from 'locks'
+const mutex = locks.createMutex();
+
+
+export default async (input) => {
+
+    const query = { name: "Budget" };
+    const update = async () => await Budget.findOneAndUpdate(query, { amount: input })
+
+
+    mutex.timedLock(5000, function (error) {
+        if (error) {
+            console.log('Could not get the lock within 5 seconds, so gave up');
+        } else {
+            console.log('We got the lock!');
+            update()
+            mutex.unlock();
+        }
```

Comment:
I like your idea of moving update to a function but I see a couple of things here that are problematic.

You are unlocking the `Budget` if everything goes right, but if an error occurs you are not unlocking the object 😱

Also, and this is not so evident if you are doing an await (performing a side effect) you could end up with an uncontrolled exception, so it is with it to surround the update function with a try/catch  statement (and don't forget to unlock the object too)